.github/workflows/publish.yml

```diff
@@ -0,0 +1,76 @@
+name: Publish
+
+on:
+  release:
+    types: [released]
+
+env:
+  NPM_TOKEN: ${{ secrets.CLPLAYERS_NPM_TOKEN_RW }}
+  VERSION: ${{ github.event.release.tag_name }}
+  WEB_PATH: "react-dom/${{ github.event.release.tag_name }}/docs/"
+
+permissions:
+  contents: read # Permission for actions/checkout
+  id-token: write # Permission for aws-actions/configure-aws-credentials
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: [ubuntu-latest]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Apply version to package.json, readme and docs
+        uses: ./.github/actions/apply-version
+        with:
+          version: ${{ env.VERSION }}
+
+    ## Publish the API docs
+
+      - name: Build Storybook
+        run: npm run build-storybook
+
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: ${{ secrets.AWS_UPLOADER_ROLE_PLAYERS }}
+          aws-region: us-east-1
+
+      - name: Upload to S3
+        run: aws s3 cp --recursive "./dist/storybook" "${{ secrets.AWS_WEB_BUCKET }}/${{ env.WEB_PATH }}"
+        # Note: the default cache policy is 1 day, I think that's OK for now.
+        # In the future if traffic increases we can increase the expiration
+        # as much as we want.
+
+    ## Publish the NPM package
+
+      - name: Write NPM RC File
+        run: |
+          echo '@castlabs:registry=https://registry.npmjs.org' > .npmrc
```

Comment:
This one gives me unease :) Can we 1) Use the secret directly here and 2) add a finally step that always executes and removes the .npmrc file?

Reply:
Yeah, I'll pass it directly to the shell that needs it.

Reply:
Actually ... I've been bamboozled. I think we both may have misunderstood what this step does - it does **_not_** write out the NPM_TOKEN to that file, it creates a templated .npmrc file, and NPM then fills that template with the current value of `$NPM_TOKEN` at runtime, meaning the token is in RAM all that time, it does not get written to a file. ... So I think we are happy with this step security-wise.

So I will keep the logic of this step, but I will move it to the `.github/actions/publish/main.js` action, which does the publishing to NPM.